src/OrchardCore.Modules/OrchardCore.Users/TimeZone/Services/UserTimeZoneSelector.cs

```diff
@@ -27,10 +27,12 @@ public async Task<TimeZoneSelectorResult> GetTimeZoneAsync()
     {
         var currentUser = await _userManager.GetUserAsync(_httpContextAccessor.HttpContext.User);
 
-        return new TimeZoneSelectorResult
-        {
-            Priority = 100,
-            TimeZoneId = async () => (await _userTimeZoneService.GetAsync(currentUser)).TimeZoneId
-        };
+        return currentUser == null
+            ? null
+            : new TimeZoneSelectorResult
+            {
+                Priority = 100,
+                TimeZoneId = async () => (await _userTimeZoneService.GetAsync(currentUser))?.TimeZoneId
+            };
```

Comment:
```suggestion
        string timeZoneId =null;
        
        if (currentUser != null)
        {
            timeZoneId  = (await _userTimeZoneService.GetAsync(currentUser))?.TimeZoneId;
        }
        return string.NullOrEmpty(timeZoneId)
            ? null
            : new TimeZoneSelectorResult
            {
                Priority = 100,
                TimeZoneId = async () => Task.FromResult(timeZoneId  )
            };
```

Reply:
This would eagerly evaluate the value then passed to `TimeZoneId`, even if it's not necessary because an `ITimeZoneSelector` with a higher priority will take precedence.

Reply:
I see that now! the LocalClock checks if the timeZoneId is IsNullOrEmpty before it uses it. so we should be good.